File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/DroneLauncher.java
package org.firstinspires.ftc.teamcode.subsystems;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.robotcore.external.Telemetry;
import org.firstinspires.ftc.teamcode.robot.Robot;
import org.firstinspires.ftc.teamcode.robot.Subsystem;

public class DroneLauncher implements Subsystem {
    private Servo droneTrigger;
    Telemetry telemetry;
    private double holdPos = 1;
    private double releasePos = 0.1;

    public DroneLauncher(Robot robot, Telemetry telemetry) {
        this.telemetry = telemetry;
        droneTrigger = robot.getServo("droneTrigger");
        hold();
    }
    public void hold(){
        droneTrigger.setPosition(holdPos);
    }
    public void release(){
        droneTrigger.setPosition(releasePos);
    }

    public void update(TelemetryPacket packet){

    }
}

File: TeamCode/src/main/kotlin/org/firstinspires/ftc/teamcode/botmodule/DroneLauncher.kt
package org.firstinspires.ftc.teamcode.botmodule

import com.qualcomm.robotcore.eventloop.opmode.OpMode
import com.qualcomm.robotcore.hardware.HardwareDevice
import com.qualcomm.robotcore.hardware.Servo

class DroneLauncher(opMode: OpMode, private var launchServo: Servo) : BotModule(opMode) {

    init {
        launchServo.position = 0.0
    }

    fun launch() {
        launchServo.position = 1.0
    }

//    public fun launchAsync(launchWithoutArming: Boolean?): Boolean { }
}